### PR TITLE
Us14 feedback

### PR DESCRIPTION
**Pull request by WildAntho**

### CODE REVIEW

File: client/src/components/Feedback/Feedback.jsx
Comment (Hamsolovski) [inline client/src/components/Feedback/Feedback.jsx:1]: C'EST LONG

**Review by Hamsolovski — APPROVED** [review]
All Good !

File: client/src/components/Card/Card.jsx [hunk 1]
```diff
@@ -1,48 +1,54 @@
...
 import { TbSunset2 } from "react-icons/tb";
 import { CiClock2 } from "react-icons/ci";
 import DarkModeContext from "../../services/DarkModeContext";
 
-
 import "./card.css";
 import CardMenu from "../CardMenu/CardMenu";
+import Feedback from "../Feedback/Feedback";
 
+// const api = import.meta.env.VITE_API_URL;
```
Comment (Alexandre78R) [inline client/src/components/Card/Card.jsx]: Delete code dead

File: client/src/components/Feedback/Feedback.jsx
Comment (Alexandre78R) [reply client/src/components/Feedback/Feedback.jsx:1]: ?

File: client/src/components/Feedback/Feedback.jsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  const details = useRef();
+
+  const handleClick = async () => {
+    if (
+      duration.current.value === "" ||
+      global.current.value === "" ||
+      difficulty.current.value === "" ||
+      after.current.value === ""
+    ) {
+      console.error("erreur");
```
Comment (Alexandre78R) [inline client/src/components/Feedback/Feedback.jsx]: delete console.error usless

File: client/src/components/Feedback/Feedback.jsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+          );
+          fetch(`${import.meta.env.VITE_API_URL}/api/trainings/${id}`, {
+            method: "PUT",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({ is_completed: 1 }),
+          });
+          if (response.ok) {
+            handleClose();
+          } else {
+            console.error("erreur client");
```
Comment (Alexandre78R) [inline client/src/components/Feedback/Feedback.jsx]: error handling please 👍

File: client/src/components/Feedback/Feedback.jsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                after: after.current.value,
+                details: details.current.value,
+                training_id: id,
+              }),
+            }
+          );
+          if (response.ok) {
+            handleClose();
+          } else {
+            console.error("erreur client");
```
Comment (Alexandre78R) [inline client/src/components/Feedback/Feedback.jsx]: error handling please 👍

File: client/src/components/Feedback/Feedback.jsx (hunk 2 above)
Comment (Alexandre78R) [reply client/src/components/Feedback/Feedback.jsx]: error handling please?

File: client/src/components/Feedback/Feedback.jsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+          }
+        }
+      } catch (error) {
+        console.error(error);
```
Comment (Alexandre78R) [inline client/src/components/Feedback/Feedback.jsx]: error handling please 👍

File: client/src/components/TipsCard/TipsCard.jsx [hunk 6]
```diff
@@ -1,16 +1,28 @@
+/* eslint-disable react/require-default-props */
```
Comment (Alexandre78R) [inline client/src/components/TipsCard/TipsCard.jsx]: No Authorized

File: client/src/pages/Journal/Journal.jsx [hunk 7]
```diff
@@ -8,54 +9,93 @@ import Days from "../../components/Days/Days";
...
       );
     } else {
       toast.info(
         "Tu n'as pas d'entraînement aujourd'hui. Profites en pour te reposer !"
       );
     }
+  };
+  useEffect(() => {
+    getToasty();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Alexandre78R) [inline client/src/pages/Journal/Journal.jsx]: Not authorized

File: server/app/controllers/feedbackActions.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    } catch(err) {
+        next(err)
+    }
+}
+
+const edit = async (req, res, next) => {
+    try {
+        const result = await tables.feedback.update(req.body, req.params.id);
+        if (result){
+            res.json(result);
```
Comment (Alexandre78R) [inline server/app/controllers/feedbackActions.js]: Please switch port edit

Comment (Alexandre78R) [reply server/app/controllers/feedbackActions.js]: ( 204 )

**Review by Alexandre78R — CHANGES_REQUESTED** [review]

**Review by Hamsolovski — APPROVED** [review]

**Review by Alexandre78R — APPROVED** [review]

**Pull request merged**